src/analyses/variable-sensitivity/variable_sensitivity_domain.cpp

```diff
@@ -449,22 +439,14 @@ void variable_sensitivity_domaint::merge_three_way_function_return(
     std::back_inserter(modified_symbols),
     [&ns](const irep_idt &id) { return ns.lookup(id).symbol_expr(); });
 
-  abstract_state = cast_function_call.abstract_state;
-  apply_domain(modified_symbols, cast_function_end, ns);
-
-  return;
-}
-
-void variable_sensitivity_domaint::apply_domain(
-  std::vector<symbol_exprt> modified_symbols,
-  const variable_sensitivity_domaint &source,
-  const namespacet &ns)
-{
   for(const auto &symbol : modified_symbols)
   {
-    abstract_object_pointert value = source.abstract_state.eval(symbol, ns);
+    abstract_object_pointert value =
+      cast_function_end.abstract_state.eval(symbol, ns);
     abstract_state.assign(symbol, value, ns);
   }
+
+  return;
```

Comment:
Seems unnecessary?

Reply:
As someone who is prone to forgetting what they are doing half way through a function and wandering off; I always like to add an explicit return, even when not strictly necessary!